tests/operators/test_operator_norm.py

```diff
@@ -0,0 +1,96 @@
+"""Tests for computing the operator norm for linear operators."""
+
+# Copyright 2024 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from math import cos
+from math import pi
+from math import prod
+
+import torch
+from mrpro.operators import EinsumOp
+from mrpro.operators import FastFourierOp
+
+from tests import RandomGenerator
+
+
+def test_correct_result():
+    """Test if the implementation yields the correct result for different choices
+    of operators with known operator-norm."""
+    random_generator = RandomGenerator(seed=0)
+
+    # test with a 3x3 matrix with known largest eigenvalue
+    matrix1 = torch.tensor([[2.0, 1, 0], [1.0, 2.0, 1.0], [0.0, 1.0, 2.0]])
+    operator1 = EinsumOp(matrix1, 'y x, x-> y')
+    random_vector1 = random_generator.float32_tensor(matrix1.shape[1])
+    operator1_norm_est = operator1.operator_norm(random_vector1, dim=None, max_iterations=32)
+    operator1_norm_true = 2 * (1 + cos(pi / 4))  # approximately 3.41421...
+    torch.testing.assert_close(operator1_norm_est.item(), operator1_norm_true, atol=1e-4, rtol=1e-4)
+
+    # test with Fast Fourier Operator (has norm 1 since norm="ortho" is used)
+    dim = (-3, -2, -1)
+    fouirer_op = FastFourierOp(dim=dim)
+
+    random_image = torch.zeros(
+        4, 4, 8, 16, dtype=torch.complex64
+    )  # also tests that the initial value is set to a random value
+    fouirer_op_norm_batched = fouirer_op.operator_norm(random_image, dim=dim, max_iterations=64)
+    fourier_op_norm_non_batched = fouirer_op.operator_norm(random_image, dim=None, max_iterations=64)
+    fourier_op_norm_true = 1.0
+
+    torch.testing.assert_close(fouirer_op_norm_batched, torch.ones(4, 1, 1, 1), atol=1e-4, rtol=1e-4)
+    torch.testing.assert_close(fourier_op_norm_non_batched.max().item(), fourier_op_norm_true, atol=1e-4, rtol=1e-4)
+    torch.testing.assert_close(fourier_op_norm_non_batched.item(), fourier_op_norm_true, atol=1e-4, rtol=1e-4)
+
+
+def test_batched_operator_norm():
+    """Test if the batched calculation of the operator-norm works on a simple
+    matrix-vector multiplication example."""
+    random_generator = RandomGenerator(seed=0)
+    input_shape = (2, 4, 8, 8)
+
+    # dimensions which define the dimensionality of the considered vector space
+    dim = (-1,)
+
+    # create a tensor to be identified as 2 * 4 (=8) 8x8 square matrices
+    matrix1 = torch.arange(prod(input_shape)).reshape(input_shape).to(torch.float32)
```

Comment:
why not random?

Reply:
if the block-diagonal matrices are entirely randomly chosen, the largest eigenvalues of the respective matrices are pretty similar. by this construction, we make sure that it is clear which of the matrices is the "dominant" one and will have the largest of all largest eigenvalues. This eliminates the possibility that the test fails due to numerical issues instead of the  function not doing what it should do.